pkg/scale/decode.go

```diff
@@ -60,7 +60,7 @@ func indirect(dstv reflect.Value) (elem reflect.Value) {
 func Unmarshal(data []byte, dst interface{}) (err error) {
 	dstv := reflect.ValueOf(dst)
 	if dstv.Kind() != reflect.Ptr || dstv.IsNil() {
-		err = fmt.Errorf("unsupported dst: %T, must be a pointer to a destination", dst)
+		err = errors.Errorf("unsupported dst: %T, must be a pointer to a destination", dst)
```

Comment:
You need to define sentinel errors (errors at global scope) and wrap them with `fmt.Errorf("%w: some context", ErrSentinel)`. Let's avoid using `github.com/pkg/errors` since it's now supported by the std library using `fmt.Errorf` and integrates nicely with the (std lib `errors`) `errors.Is` function.